src/arch/riscv/kernel/boot.c

```diff
@@ -196,7 +197,27 @@ static BOOT_CODE bool_t try_init_kernel(
     create_frames_of_region_ret_t create_frames_ret;
     create_frames_of_region_ret_t extra_bi_ret;
 
-    /* convert from physical addresses to userland vptrs */
+    /* Convert from physical addresses to userland vptrs with the parameter
+     * pv_offset, which is defined as:
+     *     virt_address + pv_offset = phys_address
+     * The offset is usually a positive value, because the virtual address of
+     * the user image is a low value and the actually physical address is much
+     * greater. However, there is no restrictions on the physical and virtual
+     * image location in general. Defining pv_offset as a signed value might
+     * seem the intuitive choice how to handle this, but there is are two
+     * catches here that break the C rules. We can't cover the full integer
+     * range then, and overflows/underflows are well defined for unsigned values
+     * only. They are undefined for signed values, even if such operations
+     * practically work in many cases due to how compiler and machine implement
+     * negative integers using the two's-complement.
+     * Assume a 32-bit system with virt_address=0xc0000000 and phys_address=0,
```

Comment:
I think the comment can be put in the commit message instead of being repeated twice for different architectures.

Reply:
I can add it there also, but my vie is that people mostly look at the code and less at the commit messages in the long run, that is why I put the comment here. One day we might be able to unify the ARM and RISC-V code here, so this exists only once, but for now RISC-V cam into existence as a clone of ARM, thus we have some duplication.